Ical.Net/DataTypes/CalDateTime.cs

```
@@ -266,35 +311,45 @@ public override int GetHashCode()
     /// Subtracts a <see cref="TimeSpan"/> from the <see cref="CalDateTime"/>.
     /// </summary>
     /// <remarks>
-    /// This will also set <seealso cref="HasTime"/> to <see langword="true"/>,
+    /// This will also add a <see cref="IDateTime.Time"/> part that did not exist before the operation,
     /// if the <see cref="TimeSpan"/> is not a multiple of 24 hours.
     /// </remarks>
     public static IDateTime operator -(CalDateTime left, TimeSpan right)
     {
         var copy = left.Copy<CalDateTime>();
-        if ((right.Ticks % TimeSpan.TicksPerDay) != 0)
+        var newValue = copy.Value - right;
+        if (!copy.HasTime && (right.Ticks % TimeSpan.TicksPerDay) == 0)
         {
-            copy.HasTime = true;
+            copy._dateOnly = DateOnly.FromDateTime(newValue);
+        }
+        else
+        {
+            copy._dateOnly = DateOnly.FromDateTime(newValue);
```

I think we shouldn't silently convert a date-only value to date-time but rather throw an exception. This is tricky in multiple ways. In particular, if DST changes. Maybe related to #660. Do we need those operators at all? If yes, we might want to be more explicit, i.e. have a 'add/subtract nominal' vs 'add/subtract accurate' in correspondence with how RFC5545 distinguishes those cases.

---

Created  #661 for tracking a separate PR